### ISSUE TITLE
-s option has no effect when -x option is used

### ISSUE THREAD

**Issue by petdance** [beyondgrep/ack3#175]
Copied from https://github.com/beyondgrep/ack2/issues/676, reported by @kawing-chiu

I'm using ack 2.24. According to the doc, `-s` should suppress error messages about nonexistent files etc. However, when file list is read from stdin, it does not work:

```
$ ack -s some_pattern not_exist
$ echo not_exist | ack -s -x some_pattern
ack: not_exist: No such file
```

**Comment by petdance** [comment]
I verified that grep does the same thing with xargs:

```
$ echo non-existent-file | xargs grep foo
grep: non-existent-file: No such file or directory
$ echo non-existent-file | xargs grep -s foo
```

**Comment by Anders-E** [comment]
The documentation states that the `-s` flag:

> Suppress error messages about nonexistent unreadable files.

However, I just noticed it does not suppress "permission denied" errors. Is this intended, and if so, do you think the docs should be clarified?

**Comment by petdance** [comment]
It should do whatever grep does in those cases, and docs should match.

**Comment by petdance** [comment]
Both #209 and #211 are PRs that aim to fix this bug. I'm looking into what each of them does, and also what's currently on my `gh175` branch.

**Comment by petdance** [comment]
Problem seems to be that it only ignores `EACCES` errors, not non-existent file errors.

        return sub {
            my $msg = shift;
            if ( $! == EACCES ) {
                return;
            }
            App::Ack::warn( $msg );
        };

**Comment by petdance** [comment]
Fixed at 93d50b8f64736268ce361805ba42ece835fa9d66.
